common/deliverclient/block_verification.go

```diff
@@ -0,0 +1,350 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package deliverclient
+
+import (
+	"bytes"
+
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric/bccsp"
+	"github.com/hyperledger/fabric/common/configtx"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/pkg/errors"
+)
+
+type CloneableUpdatableBlockVerifier interface {
+	// VerifyBlock checks block integrity and its relation to the chain, and verifies the signatures.
+	VerifyBlock(block *common.Block) error
+
+	// VerifyBlockAttestation does the same as VerifyBlock, except it assumes block.Data = nil. It therefore does not
+	// compute the block.Data.Hash() and compare it to the block.Header.DataHash. This is used when the orderer
+	// delivers a block with header & metadata only, as an attestation of block existence.
+	VerifyBlockAttestation(block *common.Block) error
```

Comment:
If it assumes the data is nil, why not pass only the header and metadata as parameters?

Reply:
just easier that way